rover-apps/bms_2022/src/BMSMonitoring.cpp

```diff
@@ -0,0 +1,130 @@
+#include "BMSMonitoring.h"
+
+#include <cmath>
+#include <map>
+
+#include "Logger.h"
+#include "mbed.h"
+
+SPI spi(SDI, SDO, CLK);  // mosi, miso, sclk
+AnalogIn currentADC(CURR_ANLG_IN);
+DigitalOut buzzer_en(BUZZER_EN);
+DigitalOut chip_select(SAMPL);
+
+BMSMonitoring::BMSMonitoring() {
+  buzzer_en.write(0);
+  chip_select.write(1);
+  spi.format(24, 3);
+  spi.frequency(1000000);
+}
+
+void BMSMonitoring::cell_monitoring() {
+  chip_select.write(0);
+
+  // TODO: recieve data from can
+  // is_cell_balancing = getRXSignal
+  // can.getRXSignalValue(HWBRIDGE::CANID::BMS, HWBRIDGE::CANSIGNAL::BMS_STATE, is_cell_balancing);
+
+  // if voltage is too low on one cell, enable balancing on that cell
+  // only way to balance is to keep all fets on so cells always have same voltage
+  int spi_message = 0b000000000000000000000000;
+  if (is_cell_balancing) {
+    spi_message = 0b111111111111111100000000;
+  }
```

Comment:
nit: can make this a ternary expression: 
`int spi_message = (is_cell_balancing ? 0b111111111111111100000000 : 0b000000000000000000000000);`